Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java

```diff
@@ -280,24 +280,25 @@ public boolean hasVariableName(String name) {
 
     @Override
     public synchronized void setVariable(String name, @Nullable Object newValue) {
-        try (PythonSnapshot fromSnapshot = takeSnapshot()) {
-            final PyDictWrapper globals = scope.mainGlobals();
-            if (newValue == null) {
-                try {
-                    globals.delItem(name);
-                } catch (KeyError key) {
-                    // ignore
-                }
-            } else {
-                if (!(newValue instanceof PyObject)) {
-                    newValue = PythonObjectWrapper.wrap(newValue);
-                }
-                globals.setItem(name, newValue);
+        // Observe any external changes that are not yet recorded
```

Comment:
Matching change needs to happen to the groovy script session, the notifyVariableChange method is called from there (and should be deleted), since it will notify the app service of a new variable... racing the current snapshot.